pkg/reconciler/vertex/scaling/client_cache.go

```diff
@@ -0,0 +1,65 @@
+package scaling
+
+import (
+	"container/list"
+	"sync"
+
+	daemonclient "github.com/numaproj/numaflow/pkg/daemon/client"
+)
+
+type entry struct {
+	address string
+	client  *daemonclient.DaemonClient
+}
+
+// ClientCache is a cache for daemon clients.
+type ClientCache struct {
```

Comment:
Why don't use a simple LRU cache?

Reply:
I think it is because he has that DeleteAll func that closes the client.

Reply:
There's a `NewWithEvict` function which can give an eviction callback, the callback is also called in `Purge()`.

```golang
// NewWithEvict constructs a fixed size cache with the given eviction
// callback.
func NewWithEvict(size int, onEvicted func(key interface{}, value interface{})) (*Cache, error) {
	lru, err := simplelru.NewLRU(size, simplelru.EvictCallback(onEvicted))
	if err != nil {
		return nil, err
	}
	c := &Cache{
		lru: lru,
	}
	return c, nil
}
```

Reply:
didn't know about this.. updated it to use simple LRU.